src/main/webapp/app/scores/exercise-scores.component.ts

```diff
@@ -71,26 +84,28 @@ export class ExerciseScoresComponent implements OnInit, OnDestroy {
     }
 
     getResults() {
-        this.resultService
+        return this.resultService
             .getResultsForExercise(this.exercise.course!.id, this.exercise.id, {
                 showAllResults: this.showAllResults,
                 ratedOnly: true,
                 withSubmissions: this.exercise.type === ExerciseType.MODELING,
                 withAssessors: this.exercise.type === ExerciseType.MODELING,
             })
-            .subscribe((res: HttpResponse<Result[]>) => {
-                const tempResults: Result[] = res.body!;
-                tempResults.forEach(result => {
-                    result.participation!.results = [result];
-                    (result.participation! as StudentParticipation).exercise = this.exercise;
-                    result.durationInMinutes = this.durationInMinutes(
-                        result.completionDate!,
-                        result.participation!.initializationDate ? result.participation!.initializationDate : this.exercise.releaseDate!,
-                    );
-                });
-                this.allResults = tempResults;
-                this.filterResults();
-            });
+            .pipe(
+                tap((res: HttpResponse<Result[]>) => {
+                    const tempResults: Result[] = res.body!;
+                    tempResults.forEach(result => {
+                        result.participation!.results = [result];
+                        (result.participation! as StudentParticipation).exercise = this.exercise;
+                        result.durationInMinutes = this.durationInMinutes(
+                            result.completionDate!,
+                            result.participation!.initializationDate ? result.participation!.initializationDate : this.exercise.releaseDate!,
+                        );
+                    });
+                    this.allResults = tempResults;
+                    this.filterResults();
+                }),
+            );
     }
 
     filterResults() {
```

Comment:
You didn't change anything here, but this method is a very good candidate for a switch statement. In this way you can put `this.results = [];` to the default block. Also please add JSDoc here

Reply:
This doesn't have anything to do with the PR.